### PR TITLE
Java CI tests

### LINKED ISSUE #252: Build tests on various distros only happens for Node client, not other clients
The Node client has some scripts to test that it is installable on 6 major distros. The other clients don't yet have that.

We should have the same sort of scripts that test installability for every client.

### PR DESCRIPTION
**Pull request by batiati**
### This PR:

- Renames the GHA from `clients-java-maven-publish.yml` to `clients-java-build.yml`, following the same naming convention used by the `go` and `node` clients.
 
- Adds CI actions for validating the Java client on various platforms and Java versions.

- Validates the binary uploaded to the GitHub Maven repository during `push` actions.

### Validation tests:
The validation test instantiates a new `Client`, assuring the JVM can load the correct platform-specific JNI code from the JAR package.

**The Java client can run on the following OS:**

- Mac OS 
- Windows
- Linux (Ubuntu)

**The Java client can run on the following JDK+Linux distros:**

- OpenJDK (Oracle Linux, Debian, and Alpine)

- Azul Zulu:  (Ubuntu, Debian, CentOS and Alpine)

- Eclipse Temurin (Ubuntu and Alpine)

- Amazon Corretto: (Amazon Linux and Alpine)

### Not covered:
The Java client can run on `aarch64`  (Mac OS and Linux), but those platforms are not natively supported by GitHub actions.

### Dependency:

It's expected this CI to fail on Mac OS and Alpine-bases distros until #268 is merged.
Closes #252

### CODE REVIEW

**Comment by batiati** [comment]
cc @eatonphil

File: .github/workflows/clients-java-build.yml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    - name: Download tigerbeetle-java package
+      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+      run: |
+        mvn dependency:copy -Dartifact=com.tigerbeetle:tigerbeetle-java:0.0.1-SNAPSHOT -DoutputDirectory=. -DrepositoryId=github
+        mkdir -p test
+        mv *.jar test/
+   
+    - name: Create a simple test program
+      run: |
+        mkdir -p test
```
Comment (sentientwaffle) [inline .github/workflows/clients-java-build.yml:105]: Would it make sense to just include this file in a "test" or "examples" directory? That way if you ever want to reproduce what CI is doing, it already exists.

File: .github/workflows/clients-java-build.yml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        java -cp "./*" Main.java
+
+  run_validation_tests_on_containers:
+    needs: prepare_validation_tests
+    runs-on: ubuntu-latest
+    name: Validate on ${{ matrix.image }}
+    strategy:
+      fail-fast: false
+      matrix:
+        image: [
```
Comment (sentientwaffle) [inline .github/workflows/clients-java-build.yml]: Is this different than
```yaml
matrix:
  image:
    - openjdk:20-oracle
    - openjdk:17-bullseye
    ...
```

File: .github/workflows/clients-java-build.yml (hunk 1 above)
Comment (batiati) [reply .github/workflows/clients-java-build.yml:105]: I wanted to test without anything from the source code, just like we can do a simple `node -e 'require(\"tigerbeetle-node\"); console.log(\"SUCCESS!\")'` (but in Java it's a little more verbose).

**Review by sentientwaffle — APPROVED** [review]

**Pull request merged**